### PR TITLE
fix OCP_OVERLY_CONCRETE_PARAMETER

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [ ] Tests written, or not not needed

### CODE REVIEW

File: src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java [hunk 1]
```diff
@@ -1603,7 +1603,7 @@ public void saveSharesInFolder(ArrayList<OCShare> shares, OCFile folder) {
      * @return
      */
     private ArrayList<ContentProviderOperation> prepareInsertShares(
-        List<OCShare> shares, ArrayList<ContentProviderOperation> operations) {
+        Iterable<OCShare> shares, ArrayList<ContentProviderOperation> operations) {
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java:1606]: @tobiasKaminsky can we change `ArrayList`here too? :)

File: src/main/java/com/owncloud/android/ui/adapter/X509CertificateViewAdapter.java [hunk 2]
```diff
@@ -214,7 +215,7 @@ private void showIssuer(X500Principal issuer, @NonNull SslUntrustedCertLayoutBin
     }
 
 
-    private Map<String, String> parsePrincipal(X500Principal principal) {
+    private Map<String, String> parsePrincipal(Principal principal) {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/adapter/X509CertificateViewAdapter.java:218]: Than we should also rename the activity 😃

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky fine, just some minor comments :)

File: src/main/java/com/owncloud/android/ui/adapter/X509CertificateViewAdapter.java (hunk 2 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/adapter/X509CertificateViewAdapter.java:218]: No everything else still is bound to 1509Cert, except from this function, which can parse any principal.

File: src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java:1606]: We cannot :/
The result of prepareInsertShares is used in getContentResover().applyBatch(…) and this requires an ArrayList.

**Comment by AndyScherzinger** [comment]
❗ rebaed waiting for the CI results before merging

**Pull request merged**